integration/test/status/test_result.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package status
+
+import (
+	"fmt"
+	"log"
+	"text/tabwriter"
+)
+
+type TestSuiteResult struct {
+	Name             string
+	TestGroupResults []TestGroupResult
+}
+
+func (r TestSuiteResult) GetStatus() TestStatus {
+	for _, result := range r.TestGroupResults {
+		if result.GetStatus() == FAILED {
+			return FAILED
+		}
+	}
+	return SUCCESSFUL
+}
+
+func (r TestSuiteResult) Print() {
+	log.Printf(">>>>>>>>>>>>>>%v<<<<<<<<<<<<<<", r.Name)
+	log.Printf(">>>>>>>>>>>>>>%v<<<<<<<<<<<<<<", string(r.GetStatus()))
+	for _, result := range r.TestGroupResults {
+		result.Print()
+	}
+	log.Printf(">>>>>>>>>>>>>>><<<<<<<<<<<<<<<")
+}
+
+type TestGroupResult struct {
+	Name        string
+	TestResults []TestResult
+}
+
+func (r TestGroupResult) GetStatus() TestStatus {
+	for _, result := range r.TestResults {
+		if result.Status == FAILED {
+			return FAILED
+		}
+	}
+	return SUCCESSFUL
+}
+
+func (r TestGroupResult) Print() {
```

Comment:
What's the difference between a test group and a test suite? From what I can gather, we will only ever have one test suite, and each group in the suite is the set of tests that get executed to validate a feature, e.g.: cpu?

Reply:
yeah that's the hope. Feel free to change the naming